components/raftstore/src/store/peer.rs

```diff
@@ -1599,6 +1620,34 @@ where
         false
     }
 
+    fn force_forward_commit_index(&mut self) {
+        let expected_alive_voter = match &self.force_leader {
+            Some(ForceLeaderState::ForceLeader {
+                expected_alive_voter,
+            }) => expected_alive_voter,
+            _ => unreachable!(),
+        };
+
+        let mut replicated_idx = self.raft_group.raft.raft_log.persisted;
+        for (peer_id, p) in self.raft_group.raft.prs().iter() {
+            let store_id = self
+                .region()
+                .get_peers()
+                .iter()
+                .find(|p| p.get_id() == *peer_id)
+                .unwrap()
+                .get_store_id();
+            if !expected_alive_voter.contains(&store_id) {
+                continue;
+            }
+            if replicated_idx > p.matched {
+                replicated_idx = p.matched;
+            }
+        }
+        self.raft_group.raft.raft_log.committed =
```

Comment:
I suggest to check if the `replicated_idx` is at current term before forcing advance the commit index.